### PR TITLE
[ROCM] enable skipped test_api cpp tests

### PR DESCRIPTION
**Pull request by hongxiayang**
[ROCM] enable skipped  test_api cpp tests

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo

### CODE REVIEW

**Comment by hongxiayang** [comment]
all rocm tests passed;
(1) InitTest tests:
```
2023-09-25T20:53:02.0912320Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::ProducesPyTorchValues_XavierUniform 
2023-09-25T20:53:02.0913240Z [gw0] [ 37%] PASSED [1.0980s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm1d 
2023-09-25T20:53:02.0914084Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm2dStateful 
2023-09-25T20:53:02.0914896Z [gw1] [ 37%] PASSED [0.9872s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::ProducesPyTorchValues_XavierUniform 
2023-09-25T20:53:02.0915711Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::ProducesPyTorchValues_XavierNormal 
2023-09-25T20:53:02.0916532Z [gw0] [ 37%] PASSED [0.9621s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm2dStateful 
2023-09-25T20:53:02.0917288Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm2dStateless 
2023-09-25T20:53:02.0918128Z [gw1] [ 37%] PASSED [0.9450s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::ProducesPyTorchValues_XavierNormal 
2023-09-25T20:53:02.0918930Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::ProducesPyTorchValues_KaimingNormal 
2023-09-25T20:53:02.0919753Z [gw0] [ 37%] PASSED [0.9538s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm2dStateless 
2023-09-25T20:53:02.0920489Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm2d 
2023-09-25T20:53:02.0921304Z [gw1] [ 37%] PASSED [0.9645s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::ProducesPyTorchValues_KaimingNormal 
2023-09-25T20:53:02.0922097Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::ProducesPyTorchValues_KaimingUniform 
2023-09-25T20:53:02.0922907Z [gw0] [ 37%] PASSED [1.0939s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm2d 
2023-09-25T20:53:02.0923650Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm3dStateful 
2023-09-25T20:53:02.0924483Z [gw1] [ 37%] PASSED [0.9616s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::ProducesPyTorchValues_KaimingUniform 
2023-09-25T20:53:02.0925288Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::CanInitializeTensorThatRequiresGrad 
2023-09-25T20:53:02.0926121Z [gw0] [ 38%] PASSED [0.9790s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm3dStateful 
2023-09-25T20:53:02.0926878Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm3dStateless 
2023-09-25T20:53:02.0927731Z [gw1] [ 38%] PASSED [1.0401s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::CanInitializeTensorThatRequiresGrad 
2023-09-25T20:53:02.0928501Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::CalculateGainWithTanh 
```
(2) IntegrationTest tests:
```
2023-09-25T20:53:02.0941351Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::IntegrationTest::CartPole 
2023-09-25T20:53:02.0942126Z [gw0] [ 39%] PASSED [1.1081s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::InstanceNorm1d 
..
2023-09-25T20:53:02.0969092Z [gw1] [ 40%] PASSED [18.4347s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::IntegrationTest::CartPole 
2023-09-25T20:53:02.0969799Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::IntegrationTest::MNIST_CUDA 
..
```
(3) ParallelTest:
```
2023-09-25T20:53:02.1797552Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ParallelTest::DataParallelUsesAllAvailableCUDADevices_CUDA 
2023-09-25T20:53:02.1798035Z [gw0] [ 99%] PASSED [1.1544s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::OperationTest::Linear_out 
2023-09-25T20:53:02.1798610Z [gw1] [100%] PASSED [1.2784s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ParallelTest::DataParallelUsesAllAvailableCUDADevices_CUDA 
2023-09-25T20:53:02.1798622Z 
```

File: test/cpp/api/CMakeLists.txt [hunk 1]
```diff
@@ -40,15 +42,10 @@ set(TORCH_API_TEST_SOURCES
   ${TORCH_API_TEST_DIR}/grad_mode.cpp
   ${TORCH_API_TEST_DIR}/operations.cpp
 )
-if(USE_CUDA)
+if(USE_CUDA OR USE_ROCM)
```
Comment (pruthvistony) [inline test/cpp/api/CMakeLists.txt:45]: The `if()` can be removed and merged to the previous `set()` statement.

**Review by pruthvistony — CHANGES_REQUESTED** [review]

File: test/cpp/api/CMakeLists.txt (hunk 1 above)
Comment (hongxiayang) [reply test/cpp/api/CMakeLists.txt:45]: > The `if()` can be removed and merged to the previous `set()` statement.

what happens for CPU only build? putting it to set() will be problem for those build

**Review by jithunnair-amd — APPROVED** [review]

**Comment by jithunnair-amd** [comment]
@malfet CI is clean. Enabling some more cpp_tests for ROCm.

**Comment by malfet** [comment]
@jithunnair-amd do you know how much longer it will take the CI to finish with those tests to add? ROCM runners were queued quite often last week. From https://hud.pytorch.org/metrics
image

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**